data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoIP2Data.java

```diff
@@ -51,18 +103,114 @@ public void switchDatabaseReader() {
      * @param attributes attributes
      * @return enriched data Map
      */
-    @Override
-    public Map<String, Object> getGeoData(InetAddress inetAddress, List<String> attributes) {
-        //TODO
-        return null;
+    public Map<String, Object> getGeoData(InetAddress inetAddress, List<String> attributes, String tempDestDir) {
+        Map<String, Object> geoData = new HashMap<>();
+        if (GeoIPProcessorService.downloadReady) {
```

Comment:
Is this right? You are using a public static variable directly here?

Reply:
Yes, the downloadReady should be static to access in without creating instance. Checking this condition because enrichment should not happen until mmdb download completes.